src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +106,211 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+            enrolledStudents, Array.from(studentEnrollRequests.values()));
+
+        this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
+          this.existingStudents = resp.students;
+          if (!this.isExistingStudentsPanelCollapsed) {
+            const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
+            this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
+          }
+          this.isExistingStudentsPresent = true;
+        });
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.enrollErrorMessage = resp.error.message;
+      },
+    });
+  }
+
+  private partitionStudentEnrollRequests(studentEnrollRequests: StudentEnrollRequest[]): void {
+    let currentStudentChunk: StudentEnrollRequest[] = [];
+    for (const request of studentEnrollRequests) {
+      currentStudentChunk.push(request);
+      if (currentStudentChunk.length >= this.numberOfStudentsPerRequest) {
+        this.allStudentChunks.push(currentStudentChunk);
+        currentStudentChunk = [];
+      }
+    }
+    if (currentStudentChunk.length > 0) {
+      this.allStudentChunks.push(currentStudentChunk);
+    }
+  }
+
+  private checkTeamsValid(studentEnrollRequests: Map<number, StudentEnrollRequest>): void {
+    const teamSectionMap: Map<string, string> = new Map();
+    const teamIndexMap: Map<string, number> = new Map();
+    const noInvalidRowsBefore: number = this.invalidRowsIndex.size;
+
+    Array.from(studentEnrollRequests.keys()).forEach((key: number) => {
+      const request: StudentEnrollRequest | undefined = studentEnrollRequests.get(key);
+      if (request === undefined) {
+        return;
+      }
+
+      if (teamSectionMap.has(request.team)) {
+        if (teamSectionMap.get(request.team) !== request.section) {
+          this.invalidRowsIndex.add(key);
+          const firstIndex: number | undefined = teamIndexMap.get(request.team);
+          if (firstIndex !== undefined) {
+            this.invalidRowsIndex.add(firstIndex);
+          }
         }
-        this.isExistingStudentsPresent = true;
-      });
+      } else {
+        teamSectionMap.set(request.team, request.section);
+        teamIndexMap.set(request.team, key);
+      }
+    });
+    if (this.invalidRowsIndex.size > noInvalidRowsBefore) {
+      this.enrollErrorMessage += 'Found duplicated teams in different sections. ';
+    }
+  }
+
+  private checkCompulsoryFields(studentEnrollRequests: Map<number, StudentEnrollRequest>): void {
+    const noInvalidRowsBefore: number = this.invalidRowsIndex.size;
+
+    Array.from(studentEnrollRequests.keys()).forEach((key: number) => {
+      const request: StudentEnrollRequest | undefined = studentEnrollRequests.get(key);
+      if (request === undefined) {
+        return;
+      }
+
+      if ((studentEnrollRequests.size >= 100 && request.section === '')
```

Comment:
Maybe about the implicit section overflow error (that size of default section exceeds 100) could be explicit to the user with a custom error message trailing behind the existing empty field one?